pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -1556,8 +1563,31 @@ public void setCharacterStream(int parameterIndex, Reader value, long length)
     throw Driver.notImplemented(this.getClass(), "setCharacterStream(int, Reader, long)");
   }
 
+  private String getString(Reader value) throws SQLException {
+    try {
+      StringBuilder v = new StringBuilder(1024);
+      char[] buf = new char[1024];
+      int nRead = 0;
+      while (nRead > -1) {
+        nRead = value.read(buf);
+        if (nRead > 1) {
+          v.append(buf, 0, nRead);
+        }
+      }
+      return v.toString();
+    } catch (IOException ioe) {
+      throw new PSQLException(GT.tr("Provided Reader failed."), PSQLState.UNEXPECTED_ERROR, ioe);
+    }
+  }
+
   public void setCharacterStream(int parameterIndex, Reader value) throws SQLException {
```

Comment:
>Larger values are streamed to a temporary file

I stay corrected.

> Given the 1GB limit on bytea columns and the 2GB length limit in the protocol, that seems like a bad idea

We can throw an appropriate "invalid argument" exception